File: v1/src/main/java/com/api/v1/book/dtos/UpdateBookRequestDto.java
package com.api.v1.book.dtos;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

public record UpdateBookRequestDto(
    @NotBlank String title,
    String subtitle,
    @NotBlank String author,
    int publishingYear,
    @NotBlank String field,
    @Min(1) int numberOfPages,
    @Min(1) int version
) {
    
}

File: kotlin/v1/src/main/kotlin/com/api/v1/book/dtos/UpdateBookRequestDto.kt
package com.api.v1.book.dtos

import jakarta.validation.constraints.*

data class UpdateBookRequestDto(
    @NotBlank val title: String,
    val subtitle: String,
    @NotBlank val publisher: String,
    @NotEmpty val publishingYear: Int,
    @Min(1) val version: Int,
    @NotEmpty val numberOfPages: Int,
    @NotBlank val author: String,
    @NotBlank val field: String
)